### ISSUE TITLE
OCR Transcriptions

### ISSUE THREAD

**Issue by DiamondSamurai0** [daledesilva/obsidian_ink#43]
**What's your idea?**
I think it would be really cool if the text you write gets converted into hidden real text that allows the hand writing to be searched 

**Why? How would it help you (or others)?**
It would encourage more handwritten notes as it would allow you to search for them in larger vaults with our having to worried about losing them

**Any examples or references you think are worth pointing out?**

**Anything else?**
This should be a toggle because it would increase file size

**Comment by daledesilva** [comment]
Yep, this is planned 🙂 and you actually described it very similar to how I plan to implement it.

I'm working to get most of the core aspects done and rock solid first, so it will be a few months at least before I get on to this.

**Comment by abishekmuthian** [comment]
@daledesilva I read else where that you're planning to implement Chat GPT API call to implement the OCR function; Kindly consider including Ollama connection as well for those of us who using Local AI; Ollama supports Chat GPT API.

**Comment by jdk-21** [comment]
I did a first implementation and it works fairly well. I've seen the Google Vision implementation, but I went with MyScripts API which is also free for the first 2000 requests. Google Vision could be added at a later date since my approach also supports different providers.
I also thought about using a local option, but this would mean that not all devices would be supported or you would need to host your own server and then connect to it. 
With the CLA and the CC license, I don't know if I should open a PR.
If anyone is interested this is the branch, I'm working on: https://github.com/jdk-21/obsidian_ink/tree/feat/handwriting-ocr

**Comment by codemaestro** [comment]
It would be cool to integrate with the [INKredible app](https://primecirca.com/inkredible) rather than a subscription—based API like MyScript. I've been using Notes Plus and INKredible since I got my first iPad 2 over 10 years ago.

**Comment by layanto** [comment]
Inkedmark obsidian plugin has feature for AI ocr (can be cloud or self hosted). I think it converts handwriting into picture and send picture to AI to transcribe.

Maybe ink plugin can gain some ideas from inkedmark plugin. I do prefer the ink plugin for embedded writing with lines and automatically adding new lines as you write.
